src/widget/wbeatmenu.cpp

```diff
@@ -0,0 +1,16 @@
+#include "wbeatmenu.h"
+
+WBeatMenu::WBeatMenu(UserSettingsPointer pConfig, QWidget* parent)
+        : QMenu(parent), m_pConfig(pConfig) {
+}
+
+void WBeatMenu::update() {
+    // TODO(hacksdump): Declare enum type for beat grid mode setting.
+    int beatMode = m_pConfig->getValue(ConfigKey("[Waveform]", "beatGridMode"), 0);
```

Comment:
I don't think WBeatMenu should be accessing configuration values... how about updating the beatgrid mode for WBeatMenu from WWaveformViewer::setBeatGridMode?